// scryfall-client/dist/index.js
"use strict";
var api_request_1 = require("./lib/api-request");
var enque_task_1 = require("./lib/api-request/enque-task");
var wrap_scryfall_response_1 = require("./lib/wrap-scryfall-response");
var convert_symbols_to_emoji_1 = require("./lib/convert-symbols-to-emoji");
var cards_1 = require("./api-routes/cards");
var sets_1 = require("./api-routes/sets");
var catalog_1 = require("./api-routes/catalog");
function setTextTransform(func) {
    wrap_scryfall_response_1.setTextTransform(func);
}
function slackify() {
    wrap_scryfall_response_1.setTextTransform(convert_symbols_to_emoji_1.slack);
}
function discordify() {
    wrap_scryfall_response_1.setTextTransform(convert_symbols_to_emoji_1.discord);
}
function resetTextTransform() {
    wrap_scryfall_response_1.resetTextTransform();
}
function setApiRequestDelayTime(waitTime) {
    enque_task_1.setTaskDelayTime(waitTime);
}
function resetApiRequestDelayTime() {
    enque_task_1.resetTaskDelayTime();
}
function getSymbolUrl(symbol) {
    var match = symbol.match(/{?(.)}?/);
    var character = match ? match[1] : symbol;
    return "https://img.scryfall.com/symbology/" + character + ".svg";
}
function wrap(body) {
    return wrap_scryfall_response_1.default(body);
}
module.exports = {
    setApiRequestDelayTime: setApiRequestDelayTime,
    resetApiRequestDelayTime: resetApiRequestDelayTime,
    setTextTransform: setTextTransform,
    slackify: slackify,
    discordify: discordify,
    resetTextTransform: resetTextTransform,
    getSymbolUrl: getSymbolUrl,
    get: api_request_1.get,
    post: api_request_1.post,
    wrap: wrap,
    getSets: sets_1.getSets,
    getSet: sets_1.getSet,
    getSetByTcgId: sets_1.getSetByTcgId,
    search: cards_1.search,
    autocomplete: cards_1.autocomplete,
    getCollection: cards_1.getCollection,
    getCards: cards_1.getCards,
    getCard: cards_1.getCard,
    getCardNamed: cards_1.getCardNamed,
    getCardBySetCodeAndCollectorNumber: cards_1.getCardBySetCodeAndCollectorNumber,
    getCatalog: catalog_1.getCatalog,
};
